codebase/src/java/disco/org/openlvc/disco/connection/rpr/RprConnection.java

```diff
@@ -580,9 +588,36 @@ public OpsCenter getOpsCenter()
 
 		// Get the create-only modules
 		ClassLoader loader = getClass().getClassLoader();
-		URL[] createModules = new URL[] {
-		    loader.getResource( "hla/rpr2/RPR-Switches_v2.0.xml" )
-		};
+		String fomPath = "hla/rpr2/RPR-Switches_v2.0.xml";
+		URL url = loader.getResource( fomPath );
+		
+		// using mak rti, so can't load modules from jar
+		if( rprConfiguration.getRtiProvider() == RtiProvider.Mak )
+		{
+			try
+			{
+				File moduleFile = new File( fomPath );
+				if( !moduleFile.exists() )
+				{
+					// open file and stream contents to new non-jar file
+					URLConnection con = url.openConnection();
+					InputStream stream = con.getInputStream();
+					Files.createDirectories( moduleFile.toPath().getParent() );
+					Files.copy( stream, moduleFile.toPath(), StandardCopyOption.REPLACE_EXISTING );
+				}
+				// change url to new non-jar file location
+				url = moduleFile.toURI().toURL();
+			}
+			catch( IOException e )
+			{
+				throw new DiscoException( "Error loading switches module for Mak", e );
+			}
+		}
+		
+		if( url == null )
+			throw new DiscoException( "Could not find FOM module: "+url );
+		
+		URL[] createModules = new URL[] { url };
```

Comment:
As per comment above, if we can move the FOM module stuff to a generic method in utils and then call it from here.